packages/dds/task-manager/src/taskManager.ts

```diff
@@ -145,16 +146,17 @@ export class TaskManagerClass
 			(taskId: string, clientId: string, local: boolean, messageId: number) => {
 				// We're tracking local ops from this connection. Filter out local ops during "connecting"
 				// state since these were sent on the prior connection and were already cleared from the latestPendingOps.
-				if (runtime.connected && local) {
+				if (local) {
 					const latestPendingOps = this.latestPendingOps.get(taskId);
-					const pendingOp = latestPendingOps?.shift();
-					assert(pendingOp !== undefined, 0x07b /* "Unexpected op" */);
-					// Need to check the id, since it's possible to volunteer and abandon multiple times before the acks
-					if (messageId === pendingOp.messageId) {
-						assert(pendingOp.type === "volunteer", 0x07c /* "Unexpected op type" */);
-						if (latestPendingOps?.length === 0) {
-							this.latestPendingOps.delete(taskId);
-						}
+					assert(latestPendingOps !== undefined, "No pending ops for task");
+					const pendingOp = latestPendingOps.shift();
+					assert(
+						pendingOp !== undefined && pendingOp.messageId === messageId,
```

Comment:
Maybe not required for this PR, but I'm realizing the `localOpMetadata as number` cast in `processCore` isn't correct for remote ops (and here, the function signature should be `messageId: number | undefined` at least).  Up to you if you want to leave it for another PR or address here, same for abandon/complete.